packages/gatsby-transformer-sqip/src/extend-node-type.js

```diff
@@ -0,0 +1,287 @@
+const crypto = require(`crypto`)
+const { extname, join, resolve } = require(`path`)
+
+const fs = require(`fs-extra`)
+const { GraphQLObjectType, GraphQLString, GraphQLInt } = require(`graphql`)
+const PQueue = require(`p-queue`)
+const ProgressBar = require(`progress`)
+const sqip = require(`sqip`)
+
+const SUPPORTED_NODES = [`ImageSharp`, `ContentfulAsset`]
+const queue = new PQueue({ concurrency: 1 })
+let bar
+
+module.exports = async args => {
+  const { type: { name } } = args
+
+  if (!SUPPORTED_NODES.includes(name)) {
+    return {}
+  }
+
+  // @todo progress bar does not show up some times
```

Comment:
This gets initialized way before the action node property resolving happens. It properly also collides with the progress of the sharp plugin